src/predicate_lang/predicate_lang.ml

```diff
@@ -94,6 +96,8 @@ and decode f = many f or_ []
 let rec encode f =
   let open Encoder in
   function
+  | True -> encode f (Or [])
```

Comment:
That seems odd: I'd expect `True -> encode f True` here.

Reply:
Did you make a typo? That would just be an infinite loop then.

It is odd, but I'm afraid it's unavoidable as we don't have a way to represent `true` or `false` as literals in the predicate language. We could introduce `:true` and `:false` for example though.

Reply:
Oh, sorry, I meant mapping `True` and `False` to the corresponding constants in the predicate language, like `:true` and `:false`, for example.

Reply:
My thinking is: if we allow `(or ())` in the language, we better also allow `(:true)` for clarity.

Reply:
Agreed, but this is actually a change that needs versioning. On the rules side, we must only accept this starting from 3.9. On the engine side, would we need to bump the action digest version because we aren't serializing `Predicate_lang.true_` in the same way?

Reply:
Right, we can do this later, of course.